core/domain/exp_fetchers.py

```diff
@@ -673,3 +676,47 @@ def get_exploration_version_history(
         version_history_model.metadata_last_edited_committer_id,
         version_history_model.committer_ids
     )
+
+
+def get_user_progress_in_exploration(
+    user_id: str, exp_ids: List[str]
+) -> List[exp_domain.ExplorationProgressSummaryDict]:
+    """Returns the progress of user in multiple exploration.
+
+    Args:
+        user_id: str. The user id of the user.
+        exp_ids: list(str). The ids of the exploration.
+
+    Returns:
+        list(ExplorationProgressSummaryDict). The list of the progress
+        summaries of the user corresponding to all exploration.
+    """
+    exp_id_to_exp_map = exp_fetchers.get_multiple_explorations_by_id(exp_ids)
+    user_id_exp_id_combinations = list(itertools.product([user_id], exp_ids))
```

Comment:
Thanks, @aasiffaizal for the review, yes we need a list that contains user_id and exp_id cartesian product so that I can fetch the data from the database using the get multi function.

Reply:
Yeah but do we need to use the list function?
I mean wouldn't it work without that since you are iterating through it down in the for loop